### PR TITLE
Feat(BCE-168): View Reports Table (Admin)

### PR DESCRIPTION
**Pull request by NickEspitalier**
JIRA: [link to jira ticket](https://champlainsaintlambert.atlassian.net/browse/BCE-168)

## Context:
This is the performance view table visible when you are connected as an admin.

## Changes

- New class model called performance chart.
- A general function to display the all the performance chart of all users when connected as an admin.

## Tests coverage

j1

### Reports menu when connected as an Admin
j2

### Reports menu when connected as a User
j3

## Dev notes (Optional)
To make development faster, add a new method in the handle refresh to display the performance chart but only as a specific user.

### CODE REVIEW

File: client/src/locales/en/translation.json [hunk 1]
```diff
@@ -129,9 +129,12 @@
         "ExportButtonLabel": "Export",
         "reports":{
             "Title":"Reports",
-            "NameLabel":"Report Name",
-            "DateLabel":"Report Date",
-            "EmployeeLabel":"Concerned Employee(s)",
```
Comment (cjayneb) [inline client/src/locales/en/translation.json:134]: Why was this removed in hoth translation files?

File: server/data_access_layer/daos/chart_report.dao.js [hunk 2]
```diff
@@ -98,4 +98,23 @@ exports.deleteChartReportById = async(chartReportId, chartReportModel = ChartRep
                 reject(response);
             });
     });
+}
+
+// Reports Types related functions
+exports.getPerformanceReportsWhenConnectedAsAdmin = async (userId, performanceReportModel = PerformanceReportModel) => {
```
Comment (cjayneb) [inline server/data_access_layer/daos/chart_report.dao.js:104]: I'm not sure why this function is in the Chart Report Dao. I think it should be in the report dao

File: server/tests/DAL/chart_report.dao.test.js [hunk 3]
```diff
@@ -374,5 +407,61 @@ describe("Test Chart Report DAO", () => {
...
+    
+                    // act and assert
+                    await expect(ChartReportDAO.getPerformanceReportsWhenConnectedAsAdmin('fakeUserId', PerformanceReportMock)).resolves
+                        .toEqual(returnedPerformanceReports);
+                });
+    
+                it("CRD5.1.2 - should resolve false when Model cant fetch data", async () => {
+                    // arrange
+                    PerformanceReportMock.$queryInterface.$useHandler(function (query, queryOptions, done) {
+                        return Promise.resolve(false);
```
Comment (cjayneb) [inline server/tests/DAL/chart_report.dao.test.js:427]: Big unecessary space

File: server/tests/DAL/report.dao.test.js [hunk 4]
```diff
@@ -5,17 +5,22 @@ const { sequelize,
...
     const RecipientInstance = new RecipientModel();
+    const PerformanceInstance = new PerformanceReportModel();
+    checkModelName(PerformanceReportModel)('performance_reports');
     checkModelName(ReportTypeModel)('report_types');
     checkModelName(RecipientModel)('recipients');
     checkModelName(ReportTypeRecipientModel)('report_type_recipients');
 
+    ['performanceReportId', 'employeeId', 'averageCollectionDay', 'annualBillingObjective', 'monthlyBillingObjective', 'annualBillingNumber', 'monthlyBillingNumber', 'projectedBonus']
+        .forEach(checkPropertyExists(PerformanceInstance));
+
```
Comment (cjayneb) [inline server/tests/DAL/report.dao.test.js:23]: This is repeated code from chart report dao test. So either put it in the report dao test or in the chart report dao test. However like I said in my other comment, I think the function to get performance reports should be in the report dao rather than the chart report dao.

File: server/tests/controllers/report.controller.test.js [hunk 5]
```diff
@@ -750,4 +750,127 @@ describe("Test Report Controller", () => {
...
+
+    describe("RC5 - getPerformanceReportsOfAllUsers", () => {
+        describe("RC5.1 - given a valid user from auth service", () => {
+            it("RC5.1.1 - should respond with 200 and body", async () => {
+                // arrange
+                let expectedResponse = [
+                    {
+                        performanceReportId: "PerformanceUUID",
+                        employeeId: 1,
+                        averageCollectionDay: "35",
```
Comment (cjayneb) [inline server/tests/controllers/report.controller.test.js:762]: Unecessary big space

**Review by cjayneb — APPROVED** [review]
Check my comments and merge :)

File: client/src/locales/fr/translation.json [hunk 6]
```diff
@@ -129,9 +129,12 @@
...
           "Title": "Rapports",
-          "NameLabel": "Nom du Rapport",
-          "DateLabel": "Date du Rapport",
-          "EmployeeLabel": "Employé(es) concerné(es)",
+          "AverageCollectionDay": "Moyenne des jours de collection",
+          "AnnualBillingObjective": "Objectif de facturation annuel",
+          "MonthlyBillingObjective": "Objectif de facturation mensuel",
+          "AnnualBillingNumber": "Numéro de facturation annuel",
+          "MonthlyBillingNumber": "Numéro de facturation mensuel",
+          "ProjectedBonus": "Bonus projeté",
```
Comment (Funnyadd) [inline client/src/locales/fr/translation.json:137]: In french, 'bonus' is 'prime'

File: server/data_access_layer/daos/chart_report.dao.js (hunk 2 above)
Comment (Funnyadd) [reply server/data_access_layer/daos/chart_report.dao.js:104]: Was going to say the same thing

**Review by Funnyadd — APPROVED** [review]
LGTM, great work in a short period of time!

File: client/src/locales/en/translation.json (hunk 1 above)
Comment (NickEspitalier) [reply client/src/locales/en/translation.json:134]: I added them back.

File: server/data_access_layer/daos/chart_report.dao.js (hunk 2 above)
Comment (NickEspitalier) [reply server/data_access_layer/daos/chart_report.dao.js:104]: Thinking about it, it makes more sense in the report dao, I'll change it now
